sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3031,7 +3031,7 @@ object GpuOverrides {
       (shuffle, conf, p, r) => new GpuShuffleMeta(shuffle, conf, p, r)),
     exec[UnionExec](
       "The backend for the union operator",
-      ExecChecks(TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64 +
+      ExecChecks(TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64 + TypeSig.MAP +
```

Comment:
This is great. Could we add in arrays too? If we don't do it now we are just going to have to add it again in a separate PR soon.

```
(TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64 + 
    TypeSig.MAP + TypeSig.STRUCT + TypeSig.ARRAY).nested()
```